airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryWriteConfig.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.bigquery;
+
+import com.google.cloud.bigquery.Schema;
+import com.google.cloud.bigquery.TableId;
+import io.airbyte.protocol.models.DestinationSyncMode;
+import java.util.ArrayList;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @param datasetId the dataset ID is equivalent to output schema
+ */
+public record BigQueryWriteConfig(
```

Comment:
there might be opportunity to DRY this too?

Reply:
This class also has many BigQuery-specific logic (e.g. `TableId`). So I'd rather not to merge it with the other write configs.